src/test/util.test.ts

```diff
@@ -69,4 +74,146 @@ describe("util", () => {
             expect(result).toEqual(`/${path}`);
         });
     });
+
+    describe("getNextAgentOffest", () => {
```

Comment:
do you have a test for what happens if you pass a bad offset?

Reply:
Nope, and don't have error handling for that either, but working on it now.

Reply:
Added some checks and tests for misshapen data that would result from bad offsets.